File: algorithms/strings/game-of-thrones/game-of-thrones-javascript/js/testeGameOfThrones.js


import {gameOfThrones} from '../js/gameOfThrones.js';
import assert from 'assert';

describe('gameOfThrones', function() {

    it('A reorganização das letras da string pode formar um palíndromo? YES', function() {

        let s = 'aaabbbb';
        let esperado = 'YES';
        let resultado = gameOfThrones(s);

        assert.deepStrictEqual(resultado, esperado);
        
    });
});

describe('gameOfThrones', function() {

    it('A reorganização das letras da string pode formar um palíndromo? NO', function() {

        let s = "cdefghmnopqrstuvw";
        let esperado = "NO";
        let resultado = gameOfThrones(s);

        assert.strictEqual(resultado, esperado);
        
    });
});

File: algorithms/strings/game-of-thrones/game-of-thrones-java/game-of-thrones-java/src/TesteGameOfThrones.java
import static org.junit.Assert.assertEquals;

import org.junit.Test;

public class TesteGameOfThrones {

    @Test
    public void testeCaso1() {

        String s = "aaabbbb";
        String esperado = "YES";

        String resultado = Resultado.gameOfThrones(s);

        assertEquals(resultado, esperado);
    }
    
    @Test
    public void testeCaso2() {

        String s = "cdefghmnopqrstuvw";
        String esperado = "NO";

        String resultado = Resultado.gameOfThrones(s);

        assertEquals(resultado, esperado);
    }
}
